azure-pipelines.yml

```diff
@@ -13,19 +13,27 @@ jobs:
       sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
       sudo apt-add-repository -y ppa:deadsnakes/ppa
       sudo apt update
-      sudo apt install -y g++-8 flex bison cmake
+      sudo apt install -y g++-8 flex bison
       sudo apt-get install -y python3.7 python3.7-dev
       python3.7 -m pip install -U pip setuptools
       python3.7 -m pip install --user 'Jinja2>=2.9.3' 'PyYAML>=3.13' pytest 'sympy>=1.3'
+      # we manually get version 3.10.2 to make sure that changes in the cmake
+      # files do not require unsupported versions of cmake in our package. 
+      wget "https://github.com/Kitware/CMake/releases/download/v3.10.2/$CMAKE_PKG.tar.gz" && tar xzpvf $CMAKE_PKG.tar.gz
+    env:
+      CMAKE_PKG: 'cmake-3.10.2-Linux-x86_64'
     displayName: 'Install Depdendencies'
   - script: |
-      export PATH=/home/vsts/.local/bin:$PATH
+      export PATH=$(pwd)/$CMAKE_PKG/bin:/home/vsts/.local/bin:$PATH
       export CXX='g++-8'
       mkdir -p $(Build.Repository.LocalPath)/build
       cd $(Build.Repository.LocalPath)/build
+      cmake --version 
       cmake .. -DPYTHON_EXECUTABLE=`which python3.7` -DCMAKE_INSTALL_PREFIX=$HOME/nmodl -DCMAKE_BUILD_TYPE=Release
       make -j 2
       make test
+    env:
+      CMAKE_PKG: 'cmake-3.10.2-Linux-x86_64'
```

Comment:
curious - what is this used for?

Reply:
This is the azure mess that I had the pleasure to deal with this morning:
azure pipeline uses ubuntu 16.04, doing `apt-get install cmake` gets you cmake 3.5.3, but the pipeline image also has the latest stable cmake installed and its path prepended, so, no matter what you install you get cmake 3.16.x . The easiest way I found to force our preferred cmake version (3.10.2) is to download and unpack the binary release from github and prepend its bin path.